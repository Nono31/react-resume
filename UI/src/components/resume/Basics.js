import React from 'react';
import PropTypes from 'prop-types';
import gravatar from 'gravatar';

const Basics = ({ resume }) => {
  if (resume.basics && resume.basics.email) {
    var gravatarPicture = gravatar.url(resume.basics.email, {
      s: '200',
      r: 'pg',
      d: 'mm'
    });
  }
  if (resume.basics.picture || gravatarPicture) {
    var profilePicture = resume.basics.picture ? resume.basics.picture : gravatarPicture;
  }

  return (
      <div className="col-sm-12">
        <div id="photo-header" className="text-center">
          <div id="photo">
            <img src={profilePicture} alt="avatar" />
          </div>
          <div id="text-header" >
            <h1>
              {resume.basics.name}
              <br />
              <span>{resume.basics.label}</span>
            </h1>
          </div>
        </div>
      </div>
  );
};

Basics.propTypes = {
  resume: PropTypes.object.isRequired
};

export default Basics;


