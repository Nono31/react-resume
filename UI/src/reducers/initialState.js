export default {
    resume: {
        "basics": {
            "location": {},
            "profiles": []
        }
    },
};
