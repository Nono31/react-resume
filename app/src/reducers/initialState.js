export default {
    resume: {
        "basics": {
            "location": {},
            "profiles": []
        },
        "work": [],
        "volunteer": [],
        "education": [],
        "awards": [],
        "publications": [],
        "skills": [],
        "languages": [],
        "interests": [],
        "references": [],
        "projects":[]
    },
};
