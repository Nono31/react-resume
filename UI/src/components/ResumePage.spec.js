import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import initialState from '../reducers/initialState';
import Basics from './resume/Basics';
import ConnectedResumePage, { ResumePage } from './ResumePage';

describe("<ResumePage />", () => {
  const actions = {};

  it("should contain <Basics />", () => {
    const wrapper = shallow(
      <ResumePage actions={actions} resume={initialState.resume} />
    );

    expect(wrapper.find(Basics).length).toEqual(1);
  });

  it("should match snapshot", () => {
    const store = configureMockStore()(initialState);
    const component = create(
      <Provider store={store}>
        <ConnectedResumePage />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
