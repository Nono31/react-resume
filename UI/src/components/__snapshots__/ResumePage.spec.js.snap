// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumePage /> should match snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="text-center"
        id="photo-header"
      >
        <div
          id="photo"
        >
          <img
            alt="avatar"
            src="//www.gravatar.com/avatar/d415f0e30c471dfdd9bc4f827329ef48?s=200&r=pg&d=mm&f=y"
          />
        </div>
        <div
          id="text-header"
        >
          <h1>
            <br />
            <span />
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-md-6 col-lg-7"
    >
      <div
        className="box"
      >
        <h2>
          <i
            className="fa fa-user ico"
          />
           About
        </h2>
        <p />
      </div>
    </div>
    <div
      className="col-12 col-md-6 col-lg-5"
    >
      <div
        className="box clearfix"
      >
        <h2>
          <i
            className="fa fa-bullseye ico"
          />
           Contact
        </h2>
      </div>
    </div>
  </div>
</div>
`;
